### PR TITLE
Augment the UMFPACK error message.

### PR DESCRIPTION
**Pull request by bangerth**
In response to a recent thread on the mailing list.

### CODE REVIEW

File: include/deal.II/lac/sparse_direct.h [hunk 1]
```diff
@@ -304,9 +304,12 @@ class SparseDirectUMFPACK : public Subscriptor
         "want to check your assembly procedure. Similarly, a "
         "matrix can be rank deficient if you forgot to apply the "
         "appropriate boundary conditions. For example, the "
-        "Laplace equation without boundary conditions has a "
-        "single zero eigenvalue and its rank is therefore "
-        "deficient by one."
+        "Laplace equation for a problem where only Neumann boundary "
+        "conditions are posed (or where you forget to apply Dirichlet "
+        "boundary conditions) has a one zero eigenvalue and its rank is "
```
Comment (drwells) [inline include/deal.II/lac/sparse_direct.h]:
```suggestion
        "boundary conditions) has exactly one eigenvalue equal to zero and its rank is "
```
"has a one zero" doesn't look right and I think its clearer to not write "one zero".

**Comment by bangerth** [comment]
@drwells -- yes, good catch!

**Comment by drwells** [comment]
This is quite picky, but are you willing to reflow the string literals so that they are all about the same length? If not then lets merge.

**Comment by bangerth** [comment]
Sure, if it makes you happy I will do that :-) Done!

**Pull request merged**